### ISSUE TITLE
Template props validation generates kebab-case key with incorrect optional modifier placement

### ISSUE THREAD

**Issue by yyx990803** [vuejs/vetur#2411]
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)
- [x] I have tried restarting VS Code or running `Vetur: Restart VLS`

## Info

- Platform:  macOS
- Vetur version: 0.28.0
- VS Code version: 1.50.1

## Problem

If a component defines optional props in camel case, e.g.

<img width="680" alt="Screen Shot 2020-10-29 at 4 13 03 PM" src="https://user-images.githubusercontent.com/499550/97627589-f5705000-1a01-11eb-8d3d-45be7bec9969.png">

Then in another component that uses it, the types generated for props validation places the optional modifier `?` incorrectly as part of the key:

<img width="703" alt="Screen Shot 2020-10-29 at 4 16 27 PM" src="https://user-images.githubusercontent.com/499550/97627702-1e90e080-1a02-11eb-8e60-4e196288f548.png">

The generated type should be

```ts
{
  'default-split'?: number
}
```

not

```ts
{
  'default-split?': number
}
```

## Reproducible Case

This can be reproduced on https://github.com/vuejs/vue-devtools/tree/next by opening https://github.com/vuejs/vue-devtools/blob/next/packages/app-frontend/src/features/components/ComponentsInspector.vue and hover over `<SplitPane>`

**Comment by yoyo930021** [comment]
Duplicate of #2314 
It will fix in #2342

**Comment by yoyo930021** [comment]
It will fix in #2342 PR.
